airflow/providers/amazon/aws/transfers/redshift_to_s3.py

```diff
@@ -124,10 +138,8 @@ def execute(self, context) -> None:
         credentials = s3_hook.get_credentials()
         credentials_block = build_credentials_block(credentials)
         unload_options = '\n\t\t\t'.join(self.unload_options)
-        s3_key = f"{self.s3_key}/{self.table}_" if self.table_as_file_name else self.s3_key
-        select_query = f"SELECT * FROM {self.schema}.{self.table}"
 
-        unload_query = self._build_unload_query(credentials_block, select_query, s3_key, unload_options)
+        unload_query = self._build_unload_query(credentials_block, self._select_query, self.s3_key, unload_options)
```

Comment:
This should fix the failing static check - https://github.com/apache/airflow/pull/14177/checks?check_run_id=1873280398#step:7:213

```suggestion
        unload_query = self._build_unload_query(
            credentials_block, self._select_query, self.s3_key, unload_options
        )
```